File: serial_comm/HeartbeatMessage.cpp
#include "HeartbeatMessage.h"

#include "Messages.h"
#include "Utils.h"

HeartbeatMessage::HeartbeatMessage() : MessageBase(Messages::HEARTBEAT)
{
  
}

Vector<unsigned char> HeartbeatMessage::encode()
{
  Vector<unsigned char> retVal(1);
  return retVal;
}
  
void HeartbeatMessage::decode(const unsigned char* data, const uint16_t size)
{
  UNUSED(data);
  UNUSED(size);
}

File: python/arduino_termpoint/comm_messaging/comm_serial/heartbeat_message.py
from ..message import Message
from .messages import Messages

class HeartbeatMessage(Message):
    def __init__(self):
        self.payload = None

    def get_msg_type(self):
        return Messages.HEARTBEAT

    def encode(self):
        return self.payload

    def decode(self, data):
        self.payload = None